lib/rails/auth/x509/filter/java.rb

```diff
@@ -1,23 +1,12 @@
-require "java"
-require "stringio"
-
 module Rails
   module Auth
     module X509
       module Filter
-        # Extract OpenSSL::X509::Certificates from Java's sun.security.x509.X509CertImpl
+        # Extract OpenSSL::X509::Certificates from java.security.cert.Certificate
         class Java
           def call(certs)
-            return unless certs
-            OpenSSL::X509::Certificate.new(extract_der(certs[0])).freeze
-          end
-
-          private
-
-          def extract_der(cert)
-            stringio = StringIO.new
-            cert.derEncode(stringio.to_outputstream)
-            stringio.string
+            return if certs.nil? || certs.empty?
+            OpenSSL::X509::Certificate.new(certs[0].get_encoded).freeze
```

Comment:
yes, it uses a more generic `java.security.cert.Certificate` abstract class to get the encoding (which has been around in Java for a looong time), rather than expect the instance of `sun.security.x509.X509CertImpl`.

Reply:
fixed it.

Reply:
Perfect, thanks!